File: Source/Rio.Models/DataTransferObjects/Offer/TradeWithMostRecentOfferDto.cs
using Rio.Models.DataTransferObjects.Account;
using Rio.Models.DataTransferObjects.WaterTransfer;
using System;

namespace Rio.Models.DataTransferObjects.Offer
{
    public class TradeWithMostRecentOfferDto
    {
        public int TradeID { get; set; }
        public string TradeNumber { get; set; }
        public AccountSimpleDto CreateUser { get; set; }
        public int OfferPostingTypeID { get; set; }
        public TradeStatusDto TradeStatus { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public DateTime OfferDate { get; set; }
        public AccountSimpleDto OfferCreateAccount { get; set; }
        public OfferStatusDto OfferStatus { get; set; }
        public AccountSimpleDto Buyer { get; set; }
        public AccountSimpleDto Seller { get; set; }
        public int TradePostingTypeID { get; set; }
        public int? WaterTransferID { get; set; }
        public WaterTransferRegistrationSimpleDto BuyerRegistration { get; set; }
        public WaterTransferRegistrationSimpleDto SellerRegistration { get; set; }
    }
}

File: Source/Rio.Web/src/app/shared/models/offer/trade-with-most-recent-offer-dto.ts
import { UserSimpleDto } from '../user/user-simple-dto';
import { TradeStatusDto } from './trade-status-dto';
import { OfferStatusDto } from './offer-status-dto';
import { WaterTransferRegistrationSimpleDto } from '../water-transfer-registration-simple-dto';
import { AccountSimpleDto } from '../account/account-simple-dto';

export class TradeWithMostRecentOfferDto {
    TradeID: number;
    TradeNumber: string;
    TradeStatus: TradeStatusDto;
    TradePostingTypeID: number;
    OfferPostingTypeID: number;
    Quantity: number;
    Price: number;
    OfferDate: Date;
    OfferCreateAccount: AccountSimpleDto;
    OfferCreatedAccountUser?: UserSimpleDto;
    OfferStatus: OfferStatusDto;
    Buyer: AccountSimpleDto;
    Seller: AccountSimpleDto;
    BuyerRegistration: WaterTransferRegistrationSimpleDto;
    SellerRegistration: WaterTransferRegistrationSimpleDto;
    WaterTransferID: number;

    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

